feat: change card video

rfcbf opened the pull request:
Video card modifications including:

- iPhone and iPad compatibility
- Use of Glass components (iOS 26)
- Card accessibility

rfcbf committed:
Replaces video card with glass card style

Updates the video list to use a glass card design, offering a modern and visually appealing aesthetic.

The new glass card includes:
- Enhanced visual elements and styling
- Adjusted layout for improved content presentation
- Uses new extension methods to correctly format the data

Also adds accessibility duration helpers for the new card style.

rfcbf committed:
Merge branch 'release/v5' into feature/change_videos

rfcbf committed:
Refactors GlassCard to improve structure

Consolidates GlassCard logic into GlassCardView for better organization and reusability.

Moves extension properties and functions into GlassCardView
from separate extensions, improving code locality.

Removes redundant GlassFavoriteButton and GlassShareButton views, using generic components for greater flexibility.

rfcbf committed:
Removes unused import

Removes unused SwiftUI import from Utils.swift.
This reduces unnecessary dependencies and improves code cleanliness.

cassio-rossi commented on MacMagazine/Features/VideosLibrary/Sources/Videos/VideosView.swift:
```diff
@@ -34,6 +34,10 @@ public struct VideosView: View {
-        .environment(\.theme, ThemeColor())
+    VideosView(
+        storage: storage,
+        favorite: .constant(false),
```
aqui tem que ser `favorite` e não `.constant(false)`

rfcbf replied:
isso é importante para o preview?

cassio-rossi replied:
mas quebra o App, certo? essa propriedade é usado para filtrar entre videos favoritados ou não. se ficar constante em false, quando o usuário quiser filtrar, não poderá ver seus videos favoritados.

rfcbf replied:
nao, nao. Isso está só no preview do Xcode.

cassio-rossi commented on .gitignore:
```diff
@@ -20,3 +20,5 @@ build
 !*.xcodeproj/**/Package.resolved
 MacMagazine/MacMagazine.xcodeproj/project.xcworkspace/xcuserdata
+
+MacMagazine/MacMagazine.xcodeproj/xcuserdata/renato_ferreira.xcuserdatad/xcdebugger/
```
sera q aqui poderíamos colocar 2 níveis antes?
`MacMagazine/MacMagazine.xcodeproj/xcuserdata/`
assim prevenimos qq usuário de adicionar suas próprias configurações

rfcbf replied:
Feito.

cassio-rossi commented on MacMagazine/Features/VideosLibrary/Sources/Videos/Card/GlassCardView.swift:
```diff
@@ -0,0 +1,285 @@
+
+
+@MainActor
+public struct GlassCard: VideoCard {
```
precisa atualizar aqui para refletir o novo modelo

rfcbf replied:
feito

cassio-rossi commented on MacMagazine/Features/VideosLibrary/Sources/Videos/Card/GlassCardView.swift:
```diff
@@ -0,0 +1,285 @@
+                        .tint(.primary)
+
+                }
+                .glassEffectUnion(id: 1, namespace: namespace)
```
não há a necessidade de namespace já que está tudo dento do mesmo container.

rfcbf replied:
o namespace é obrigatório para o uso do .glassEffectUnion

rfcbf committed:
Updates project configuration and UI

Adds a comprehensive .gitignore file, improving project hygiene
and preventing unnecessary files from being tracked in version control.

Updates the GlassCardView to enhance UI by using the
MacMagazineLibrary and improves the code structure.

rfcbf committed:
Merge branch 'release/v5' into feature/change_videos

rfcbf committed:
Merge branch 'release/v5' into feature/change_videos
